content/en/docs/instrumentation/go/sampling.md

```diff
@@ -1,36 +1,45 @@
-in general you should make a decision at the start of a trace, and allow the
+Sampling is a process that restricts the amount of spans that are generated by a
+system. The exact sampler you should use depends on your specific needs, but in
+general you should make a decision at the start of a trace, and allow the
 sampling decision to propagate to other services.
 
-A sampler needs to be set on the tracer provider when its configured, as
-follows:
+A [`Sampler`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace#Sampler) can
+be set on the tracer provider using the
+[`WithSampler`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace#WithSampler)
+option, as follows:
 
 ```go
-provider := sdktrace.NewTracerProvider(
-    sdktrace.WithSampler(sdktrace.AlwaysSample()),
+provider := trace.NewTracerProvider(
+    trace.WithSampler(trace.AlwaysSample()),
 )
 ```
 
-`AlwaysSample` and `NeverSample` are fairly self-explanatory. Always means that
-every trace will be sampled, the converse holds as true for Never. When you're
-getting started, or in a development environment, you'll almost always want to
-use `AlwaysSample`.
+[`AlwaysSample`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace#AlwaysSample)
+and
+[`NeverSample`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace#NeverSample)
+are self-explanatory values. `AlwaysSample` means that every span is sampled,
+while `NeverSample` means that no span is sampled. When you're getting started,
+or in a development environment, use `AlwaysSample`.
 
 Other samplers include:
 
-- `TraceIDRatioBased`, which will sample a fraction of traces, based on the
-  fraction given to the sampler. Thus, if you set this to .5, half of traces
-  will be sampled.
-- `ParentBased`, which behaves differently based on the incoming sampling
-  decision. In general, this will sample spans that have parents that were
-  sampled, and will not sample spans whose parents were _not_ sampled.
+- [`TraceIDRatioBased`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace#TraceIDRatioBased),
+  which samples a fraction of spans, based on the fraction given to the sampler.
+  If you set .5, half of all the spans are sampled.
+- [`ParentBased`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace#ParentBased),
+  is a sampler decorator which behaves differently, based on the parent of the
+  span. If the span has no parent, the decorated sampler is used to make
+  sampling decision based on the parent of the span. By default, `ParentBased`
+  samples spans that have parents that were sampled, and doesn't sample spans
+  whose parents were not sampled.
 
-When you're in production, you should consider using the `TraceIDRatioBased`
-sampler with the `ParentBased` sampler.
+By default, the tracer provider uses a `ParentBased` sampler with the
+`AlwaysSample` sampler.
+
+When in a production environment, consider using the `ParentBased` sampler with
+the `TraceIDRatioBased` sampler.
```

Comment:
I'm thinking we may want to revise this, since it's a little complicated.

If you combine both, then yes, it will typically behave how you like. However, if your application is called by an upstream one you don't necessarily control, then it can be advantageous to omit the `ParentBased` composite sampler. The result is that you can sometimes end up with traces that have missing root spans, but the rest of your traces will be coherent and at the volume you care about. We've run into this several times with customers who do head sampling and it unfortunately means they have to pick one or the other. We've tried to get them to switch to tail-based sampling but they often can't.

I'm thinking of rolling that guidance out across all the languages, but if you agree, would you want to incorporate here, or handle orthogonally?

Reply:
I am not sure if I follow. The `ParentBased` is configurable regarding local vs remote parent spans. See the available options https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace#ParentBasedSamplerOption

> I'm thinking of rolling that guidance out across all the languages, but if you agree, would you want to incorporate here, or handle orthogonally?

We can try incorporating it here. If we will have problems doing so then we can handle it orthogonally.

Reply:
Okay, sounds good. We can tackle separately.